kernel/kernel-api/src/test/scala/io/delta/kernel/internal/tablefeatures/TableFeaturesSuite.scala

```diff
@@ -18,18 +18,165 @@ package io.delta.kernel.internal.tablefeatures
 import io.delta.kernel.data.{ArrayValue, ColumnVector, MapValue}
 import io.delta.kernel.exceptions.KernelException
 import io.delta.kernel.internal.actions.{Format, Metadata, Protocol}
-import io.delta.kernel.internal.tablefeatures.TableFeatures.validateWriteSupportedTable
+import io.delta.kernel.internal.tablefeatures.TableFeatures.{TABLE_FEATURES, validateWriteSupportedTable}
 import io.delta.kernel.internal.util.InternalUtils.singletonStringColumnVector
+import io.delta.kernel.internal.util.VectorUtils.stringVector
 import io.delta.kernel.types._
 import org.scalatest.funsuite.AnyFunSuite
 
+import java.util.stream.Collectors
+import java.util.stream.Collectors.toList
 import java.util.{Collections, Optional}
 import scala.collection.JavaConverters._
 
 /**
  * Suite that tests Kernel throws error when it receives a unsupported protocol and metadata
  */
 class TableFeaturesSuite extends AnyFunSuite {
+
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+  // Tests for [[TableFeature]] implementations                                                  //
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+  val readerWriterFeatures = Seq("columnMapping", "deletionVectors", "timestampNtz",
```

Comment:
Q: would it be helpful to define these in `TableFeatures.java`. Do we see other areas of the code wanting to know these groups?

Reply:
I think I hard coded here, because we know these are things we are looking for support. If we add them in `TableFeatures.java` we are using the inputs from the prod code to test the prod.

`TableFeatures` has a list already which if needed in the prod can use it.